catroid/src/main/java/org/catrobat/catroid/web/CatrobatServerCalls.kt

```diff
@@ -0,0 +1,198 @@
+import org.catrobat.catroid.web.ServerAuthenticationConstants.FILE_TAG_URL_HTTP
+import org.catrobat.catroid.web.ServerAuthenticationConstants.JSON_ANSWER
+import org.catrobat.catroid.web.ServerAuthenticationConstants.JSON_STATUS_CODE
+import org.catrobat.catroid.web.ServerAuthenticationConstants.OAUTH_TOKEN_AVAILABLE
+import org.catrobat.catroid.web.ServerAuthenticationConstants.SERVER_RESPONSE_TOKEN_OK
+import org.catrobat.catroid.web.ServerAuthenticationConstants.SIGNIN_EMAIL_KEY
+import org.catrobat.catroid.web.ServerAuthenticationConstants.SIGNIN_OAUTH_ID_KEY
+import org.catrobat.catroid.web.ServerAuthenticationConstants.SIGNIN_USERNAME_KEY
+import org.catrobat.catroid.web.ServerAuthenticationConstants.USERNAME_AVAILABLE
+import org.json.JSONException
+import org.json.JSONObject
+import java.util.HashMap
+
+class CatrobatServerCalls(private val okHttpClient: OkHttpClient = CatrobatWebClient.client) {
+    private val tag = CatrobatServerCalls::class.java.simpleName
+
+    private var resultString: String? = null
+
+    @Throws(WebconnectionException::class)
+    fun checkToken(token: String, username: String, baseUrl: String): Boolean {
+        try {
+            val postValues = HashMap<String, String>()
+            postValues[Constants.TOKEN] = token
+            postValues[SIGNIN_USERNAME_KEY] = username
+
+            val serverUrl = baseUrl + CHECK_TOKEN_URL
+
+            resultString = WebCallBuilder(serverUrl, postValues).performCall(okHttpClient)
+
+            val jsonObject = JSONObject(resultString)
+            val statusCode = jsonObject.getInt(JSON_STATUS_CODE)
+            val serverAnswer = jsonObject.optString(JSON_ANSWER)
+
+            return if (statusCode == SERVER_RESPONSE_TOKEN_OK) {
+                true
+            } else {
+                throw WebconnectionException(statusCode, "server response token ok, but error: $serverAnswer")
+            }
+        } catch (e: JSONException) {
+            throw WebconnectionException(WebconnectionException.ERROR_JSON, Log.getStackTraceString(e))
+        }
+    }
+
+    @Throws(WebconnectionException::class)
+    private fun getRequest(url: String): String = WebCallBuilder(url, HashMap()).performCall(okHttpClient)
+
+    fun getTags(language: String?): String {
+        return try {
+            var serverUrl = FILE_TAG_URL_HTTP
+            if (language != null) {
+                serverUrl += "?language=$language"
+            }
+            getRequest(serverUrl)
+        } catch (e: WebconnectionException) {
+            Log.e(tag, Log.getStackTraceString(e))
+            ""
+        }
+    }
+
+    @Throws(WebconnectionException::class)
+    fun checkOAuthToken(id: String, oauthProvider: String, context: Context?): Boolean? {
+        try {
+            val postValues = HashMap<String, String>()
+            postValues[SIGNIN_OAUTH_ID_KEY] = id
+
+            val serverUrl: String
+            when (oauthProvider) {
```

Comment:
try to use `when` as an expression here, i.e. `val serverUrl = when (oauthProvider) {....}